File: src/main/java/com/machfour/macros/core/FoodType.java
package com.machfour.macros.core;

import java.util.NoSuchElementException;

public enum FoodType {
    PRIMARY("primary"), COMPOSITE("composite"), USDA("usda"), NUTTAB("nuttab");

    final String name;

    FoodType(String name) {
        this.name = name;
    }

    public static FoodType fromString(String name) {
        switch (name) {
            case "primary":
                return PRIMARY;
            case "composite":
                return COMPOSITE;
            case "usda":
                return USDA;
            case "nuttab":
                return NUTTAB;
            default:
                throw new NoSuchElementException("No FoodType with name '" + name + "'.");
        }
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name;
    }

}

File: src/main/kotlin/com/machfour/macros/core/FoodType.kt
package com.machfour.macros.core

enum class FoodType(val niceName: String, val defaultSearchRelevance: SearchRelevance) {
    PRIMARY("primary", SearchRelevance.USER),
    COMPOSITE("composite", SearchRelevance.USER),
    USDA("usda", SearchRelevance.INBUILT),
    NUTTAB("nuttab", SearchRelevance.INBUILT),
    SPECIAL("special", SearchRelevance.INBUILT);

    override fun toString(): String {
        return niceName
    }
    
    companion object {
        fun fromString(name: String): FoodType? {
            return when (name) {
                "primary" -> PRIMARY
                "composite" -> COMPOSITE
                "usda" -> USDA
                "nuttab" -> NUTTAB
                "special" -> SPECIAL
                else -> null
            }
        }
    }

}
